### ISSUE TITLE
Uncaught TypeError: Elm.fullscreenDebug is not a function when running the tutorial application in debug mode

### ISSUE THREAD

**Issue by alexspurling** [elm-lang/elm-reactor#131]
I'm just starting Elm. I followed the instructions on the Elm Architecture Tutorial. I cloned [its repository](https://github.com/evancz/elm-architecture-tutorial/) (at 040e412897), opened example 1, ran elm-reactor and navigated to http://localhost:8000/Counter.elm?debug to be met with a blank page. The JS console reveals the error in the issue title. The elm-reactor console itself shows:

```
Elm Reactor 0.3.2 (Elm Platform 0.15.1)
Listening on http://0.0.0.0:8000/
```

**Comment by vilterp** [comment]
Unable to reproduce this. Do you see `<script src="/_reactor/debug.js"></script>` in the DOM somewhere? That's where `Elm.fullscreenDebug` is defined

**Comment by alexspurling** [comment]
The debug.js script is included. However, it looks like it fails to run. When I navigate to: http://localhost:8000/_reactor/debug.js it shows:

```
A web handler threw an exception. Details:
user error (Unable to find the ELM_HOME environment variable when searching
for the _reactor/debug.js file.

If you installed Elm Platform with the Mac or Windows installer, it looks like
ELM_HOME was not set automatically. Look up how to set environment variables
on your platform and set ELM_HOME to the directory that contains Elm's static
files:

  * On Mac it is /usr/local/share/elm
  * On Windows it is one of the following:
      C:/Program Files/Elm Platform/0.15.1/share
      C:/Program Files (x86)/Elm Platform/0.15.1/share

If it seems like a more complex issue, please report it here:
    <https://github.com/elm-lang/elm-platform/issues>
)
```

I am not running OSX or Windows, I'm running Linux Mint. I installed the elm platform via `npm install -g elm` however, it does seem that my ELM_HOME environment variable is indeed unset. I will try to set it if I can figure out what it should be set to.

**Comment by jvoigtlaender** [comment]
The [README of the npm installer](https://www.npmjs.com/package/elm) tells you what the `ELM_HOME` variable should be set to. And due to [this commit](https://github.com/elm-lang/elm-compiler/commit/8972f1e59b644f33784f541e3f4dde1c4d09e963), the next release of Elm would also have told you as part of the error message you saw above.

**Comment by alexspurling** [comment]
Setting ELM_HOME to "/usr/local/lib/node_modules/elm/" works. It would probably have helped if I knew to read the npm installer README. However, I believe the error displayed when loading debug.js is not helpful because it requires special knowledge to actually see it. I believe it would be sensible to check the status of the ELM_HOME variable in elm-reactor itself and display a warning on the console if it is unset.

**Comment by edbond** [comment]
setting
`export ELM_HOME=/usr/lib/node_modules/elm/share`
helped me too.

**Comment by jazmit** [comment]
For me (on Ubuntu 15.04) the following worked:

**Comment by wildlyinaccurate** [comment]
If you use NVM, y `ELM_HOME` can be set to something like `~/.nvm/versions/node/$(node -v)/lib/node_modules/elm/share`

**Comment by netaisllc** [comment]
@jvoigtlaender I'm with @alexspurling 100%, the error message text ...

> set ELM_HOME to the directory that contains Elm's static files:

is unhelpful for two reasons:

1. It assumes the reader knows what **Elm's static files** are; trust me, we don't. :)
2. It ignores the *Nix world.

I don't know enough yet to offer a solution, but will keep trying.

**Comment by jvoigtlaender** [comment]
@netaisllc, basically you need to look, on your machine, for a directory that contains a subdirectory structure `reactor/_reactor/`. So if your machine has some directory `whatever/path/reactor/_reactor/`, then `ELM_HOME` should be set to `whatever/path`. It's not really possible to say anything more specific, given how varied different operating systems and install methods are.

Also, I'm not sure in what sense this ignores the *Nix world. If you are on *Nix, then you either installed from source or via npm. If you installed from source, then you shouldn't have to set `ELM_HOME` at all. If you installed via npm, then why is the hint that the error message is giving you ("you have to set ELM_HOME to a certain directory of the form `.../node_modules/elm/share`") not enough? Without knowing more about where npm lives on your machine, the error message cannot say more.

In any case, the plan for Elm 0.16, which will be released in the next few weeks, is that setting `ELM_HOME` will not be necessary any more, since the static resources will be compiled into the executables.

**Comment by netaisllc** [comment]
Yeah I understand what you are saying, the variety of places that NodeJs can end up is astounding.

**Comment by fooblahblah** [comment]
I'm running NixOS and have the following shell definition working out well (I'm using a user based global install for npm. https://github.com/sindresorhus/guides/blob/master/npm-global-without-sudo.md)

```nix
with import <nixpkgs> {}; let
  runtimeLibs = [ gmp nodejs ncurses saw-tools zlib ];
  libPaths    = map (x: ":${x}/lib") runtimeLibs;
in rec {
  ElmEnv = stdenv.mkDerivation {
    name        = "elm";
    buildInputs = [ stdenv ];
    LD_LIBRARY_PATH=lib.foldl (x: y: x + y) "" libPaths;
    ELM_HOME=~/.npm-packages/lib/node_modules/elm/share;
  };
}
```

**Comment by fooblahblah** [comment]
Oh and I had to patchelf the installed elm bits after running npm install. I'd like to create a proper install package, but haven't done so yet. Probably veering off topic for this issue...

**Comment by kurtharriger** [comment]
This fix doesn't appear to be working for me.
I installed node via homebrew on OS X so after some digging I found I should set ELM_HOME as follows and verified that I could find the desired debug file:

```
% export ELM_HOME=/usr/local/lib/node_modules/elm/Elm-Platform/0.16.0/share/
% ls $ELM_HOME/reactor/_reactor/debug.js
/usr/local/lib/node_modules/elm/Elm-Platform/0.16.0/share//reactor/_reactor/debug.js
```

However I still get the error after restarting elm reactor.
```
_U.list is not a function


Open the developer console for more details.
````

I can also navigate to http://localhost:8000/_reactor/debug.js and view the js file just fine (both before and after setting ELM_HOME).  

Is there something else I should check for?
